### ISSUE TITLE
Limit for data length?

### ISSUE THREAD

**Issue by CrossNox** [welch/seasonal#8]
Hi, this code runs fine:

```
import math
import numpy as np
from seasonal import fit_seasons, adjust_seasons
import matplotlib.pyplot as plt
length = 318
s = [10 * math.sin(i * 2 * math.pi / 25) + i * i /100.0 for i in range(length)]
seasons, trend = fit_seasons(s)
adjusted = adjust_seasons(s, seasons=seasons)
residual = adjusted - trend
plt.figure()
plt.plot(s, label='data')
plt.plot(trend, label='trend')
plt.plot(residual, label='residual')
plt.legend(loc='upper left')
plt.show()
```

However with a value of ```length = 319```, i get ```TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'```, why would that be?

**Comment by welch** [comment]
TLDR: you should check for None in the returns from fit_seasons or adjust_seasons, it means no seasonal component was detected.

You're fitting a seasonal + trend model to data that is generated by a different process (you're summing a low-amplitude sine wave of increasing frequency and a parabola). For short sequences, a linear trend is a decent fit to your quadratic term, "decent" meaning that after subtracting it from the data we can resolve the underlying sine wave as a seasonal component and cannot rule out that the frequency is constant but noisy. But longer sequences diverge enough from linear that their residuals swamp the amplitude of your sine wave, whose frequency has begun changing enough, that the correct answer (No seasonality) is at last returned.
